src/audio.c

```diff
@@ -629,6 +661,88 @@ static void check_telev(struct audio *a, struct autx *tx)
 }
 
 
+static bool silence(const void *sampv, size_t sampc, int fmt)
+{
+	const int16_t *v;
+
+	if (fmt != AUFMT_S16LE)
+		return true;
+
+	v = sampv;
+	int32_t sum = 0;
+	size_t i;
+
+	for (i = 0; i < sampc; i++) {
+		sum += v[i]*v[i];
+
+		if (sum > (int32_t) (i + 1) * SILENCE_Q)
+			return false;
+	}
+
+	return true;
+}
+
+
+static void audio_decode(void *arg)
+{
+	struct audio *a = arg;
+	struct aurx *rx = &a->rx;
+	int err = 0;
+
+	while (rx->wcnt > 0) {
+		err = stream_decode(a->strm);
+		rx->wcnt--;
+	}
+
+	while (err == EAGAIN ||
+			(!err && aubuf_cur_size(rx->aubuf) < rx->num_bytes)) {
```

Comment:
I have segfault with call hangup:
```
#0  0x00007fc9c2d02a76 in pthread_rwlock_rdlock () at /usr/lib/libpthread.so.0
#1  0x000055bff209f1b0 in lock_read_get (l=0xb5b5b5b5b5b5b5b5) at src/lock/rwlock.c:66
#2  0x000055bff20add0a in aubuf_cur_size (ab=0x55bff2c78d80) at src/aubuf/aubuf.c:339
#3  0x000055bff1fe5c35 in audio_decode (arg=0x55bff2c695f0) at src/audio.c:698
#4  0x000055bff1fe5d70 in rx_thread (arg=0x55bff2c695f0) at src/audio.c:738
#5  0x00007fc9c2cfe3e9 in start_thread () at /usr/lib/libpthread.so.0
#6  0x00007fc9bf162293 in clone () at /usr/lib/libc.so.6
```

Reply:
Tried hangup on both sides. For the moment I am not able to reproduce this. It seems that the decoding thread was not stopped cleanly. I should replace two lines in stop_rx(). Like this:
```
		rx->thr.run = false;
		pthread_cond_signal(&rx->thr.cond);
```

Reply:
Looks good now. Thanks.

Reply:
Ok, also saw the segfault if it was an outgoing call.